src/main/java/uk/gov/companieshouse/filinghistory/api/repository/Repository.java

```diff
@@ -39,24 +39,30 @@ public Repository(final MongoTemplate mongoTemplate) {
 
     public FilingHistoryListAggregate findCompanyFilingHistory(String companyNumber,
             int startIndex, int itemsPerPage, List<String> categories) {
+        try {
+            Criteria criteria = Criteria.where("company_number").is(companyNumber);
+            if (!categories.isEmpty()) {
+                criteria.and("data.category").in(categories);
+            }
 
-        Criteria criteria = Criteria.where("company_number").is(companyNumber);
-        if (!categories.isEmpty()) {
-            criteria.and("data.category").in(categories);
-        }
-
-        Aggregation aggregation = newAggregation(
-                match(criteria),
-                facet(
-                        count().as("count")).as("total_count")
-                        .and(match(new Criteria()), sort(Direction.DESC, "data.date")).as("document_list"),
-                unwind("$total_count", true),
-                project()
-                        .and(ifNull("$total_count.count").then(0)).as("total_count")
-                        .andExpression("$document_list").slice(itemsPerPage, startIndex).as("document_list"));
+            Aggregation aggregation = newAggregation(
+                    match(criteria),
+                    facet(
+                            count().as("count")).as("total_count")
+                            .and(match(new Criteria()), sort(Direction.DESC, "data.date")).as("document_list"),
+                    unwind("$total_count", true),
+                    project()
+                            .and(ifNull("$total_count.count").then(0)).as("total_count")
+                            .andExpression("$document_list").slice(itemsPerPage, startIndex).as("document_list"));
 
-        return mongoTemplate.aggregate(aggregation, FilingHistoryDocument.class, FilingHistoryListAggregate.class)
-                .getUniqueMappedResult();
+            return mongoTemplate.aggregate(aggregation, FilingHistoryDocument.class, FilingHistoryListAggregate.class)
+                    .getUniqueMappedResult();
+        } catch (DataAccessException ex) {
+            LOGGER.error("MongoDB unavailable when locating documents for company number: %s with message: %s"
```

Comment:
Log message is a bit verbose - we dont need to include company number in the message as its already in the logMap.

Reply:
Changed in latest commit to be same as the exception message.